test/rhsm/unit/test_profile.py

```diff
@@ -33,6 +33,14 @@ def setUp(self) -> None:
         self.libdnf_mock = libdnf_patcher.start()
         self.addCleanup(libdnf_patcher.stop)
 
+        # Make the following method no-op. Otherwise, it may cause
+        #   TypeError: 'Mock' object is not iterable
+        # The error is non-deterministic and only occurs as a result of mocking somewhere else.
+        # We weren't able to find the original cause.
```

Comment:
i'd mention here what was the issue we ran into, ie that for some reason the detected cloud provider in `fix_aws_rhui_repos()` is "aws", causing that cloud provides to be used (failing)